packages/amplify-graphql-schema-generator/src/schema-generator/schema-overrides.ts

```diff
@@ -1,15 +1,17 @@
 import { DocumentNode, FieldDefinitionNode, ObjectTypeDefinitionNode, visit } from 'graphql';
 import { isArrayOrObject, findMatchingField, getNonModelTypes, isOfType, isNonNullType } from 'graphql-transformer-common';
 import { printer } from '@aws-amplify/amplify-prompts';
+import { FieldWrapper, ObjectDefinitionWrapper } from '@aws-amplify/graphql-transformer-core';
 
 export const applySchemaOverrides = (document: DocumentNode, existingDocument?: DocumentNode | undefined): DocumentNode => {
   if (!existingDocument) {
     return document;
   }
 
+  let updatedDocument = preserveRelationalDirectives(document, existingDocument) as any;
```

Comment:
One alternative here is to extend the schema visitor pattern we use for `FieldDefinition` to `ObjectTypeDefinition` as well, this could save us from multiple mutations of `updatedDocument`. Non-blocking suggestion.